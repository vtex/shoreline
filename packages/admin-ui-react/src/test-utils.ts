import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
