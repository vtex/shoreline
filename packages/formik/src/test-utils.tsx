import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
