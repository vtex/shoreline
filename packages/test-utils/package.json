{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
