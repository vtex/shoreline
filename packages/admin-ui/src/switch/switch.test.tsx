import React from 'react'
import { render, axe } from '../test-utils'

import type { SwitchStateReturn } from './types'
import { Switch, useSwitchState } from './index'

function SwitchState({
  children,
}: {
  children: (state: SwitchStateReturn) => JSX.Element
}) {
  const state = useSwitchState()

  return children(state)
}

describe('Switch tests', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(
      <SwitchState>
        {(state) => (
          <Switch
            id="switch"
            state={state}
            value="toggle"
            aria-label="toggle"
            label="Label"
          />
        )}
      </SwitchState>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
