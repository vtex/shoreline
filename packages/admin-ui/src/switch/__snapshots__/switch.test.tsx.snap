// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch tests all variants should match snapshot disabled 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-3 >* {
  margin-left: 0.5rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-2 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-0 {
  background: #B9B9B9;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  cursor: pointer;
  margin: unset;
  border-radius: 6.25rem;
  height: 1.25rem;
  width: 2.25rem;
}

.emotion-0:hover {
  background: #747474;
}

.emotion-0:active {
  background: #38853C;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
  background: #747474;
}

.emotion-0:focus:hover {
  background: #747474;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]):hover {
  background: #747474;
}

.emotion-0:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}

.emotion-0:checked {
  background: #38853C;
}

.emotion-0:checked:hover {
  background: #285F2B;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #ECF7EC,0rem 0rem 0rem 0.1875rem #83D187;
  background: #285F2B;
}

.emotion-0:checked:focus:hover {
  background: #285F2B;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #38853C;
}

.emotion-0:checked:focus:not([data-focus-visible-added]):hover {
  background: #285F2B;
}

.emotion-0:checked:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}



.emotion-0:checked:checked:after {
  -webkit-transform: translateX(1rem);
  -moz-transform: translateX(1rem);
  -ms-transform: translateX(1rem);
  transform: translateX(1rem);
}



.emotion-0:checked:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-0:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:after {
  background: #FFFFFF;
  content: "";
  display: block;
  position: absolute;
  border-radius: 1.25rem;
  -webkit-transition: -webkit-transform .25s ease;
  transition: transform .25s ease;
}

<div
    class="emotion-4"
  >
    <div
      class="emotion-3"
    >
      <input
        aria-checked="false"
        aria-disabled="true"
        aria-label="label2"
        class="emotion-0"
        disabled=""
        id="switch"
        role="switch"
        style="pointer-events: none;"
        type="checkbox"
      />
      <div
        class="emotion-2"
      >
        <label
          class="emotion-1"
          for="switch"
        >
          Label
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Switch tests all variants should match snapshot label only 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-3 >* {
  margin-left: 0.5rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-2 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-0 {
  background: #B9B9B9;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  cursor: pointer;
  margin: unset;
  border-radius: 6.25rem;
  height: 1.25rem;
  width: 2.25rem;
}

.emotion-0:hover {
  background: #747474;
}

.emotion-0:active {
  background: #38853C;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
  background: #747474;
}

.emotion-0:focus:hover {
  background: #747474;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]):hover {
  background: #747474;
}

.emotion-0:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}

.emotion-0:checked {
  background: #38853C;
}

.emotion-0:checked:hover {
  background: #285F2B;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #ECF7EC,0rem 0rem 0rem 0.1875rem #83D187;
  background: #285F2B;
}

.emotion-0:checked:focus:hover {
  background: #285F2B;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #38853C;
}

.emotion-0:checked:focus:not([data-focus-visible-added]):hover {
  background: #285F2B;
}

.emotion-0:checked:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}



.emotion-0:checked:checked:after {
  -webkit-transform: translateX(1rem);
  -moz-transform: translateX(1rem);
  -ms-transform: translateX(1rem);
  transform: translateX(1rem);
}



.emotion-0:checked:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-0:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:after {
  background: #FFFFFF;
  content: "";
  display: block;
  position: absolute;
  border-radius: 1.25rem;
  -webkit-transition: -webkit-transform .25s ease;
  transition: transform .25s ease;
}

<div
    class="emotion-4"
  >
    <div
      class="emotion-3"
    >
      <input
        aria-checked="false"
        aria-label="toggle"
        class="emotion-0"
        id="switch"
        role="switch"
        type="checkbox"
        value="toggle"
      />
      <div
        class="emotion-2"
      >
        <label
          class="emotion-1"
          for="switch"
        >
          Label
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Switch tests all variants should match snapshot with error text 1`] = `
<DocumentFragment>
  .emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-6 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-5 >* {
  margin-left: 0.5rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-0 {
  background: #B9B9B9;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  cursor: pointer;
  margin: unset;
  border-radius: 6.25rem;
  height: 1.25rem;
  width: 2.25rem;
}

.emotion-0:hover {
  background: #747474;
}

.emotion-0:active {
  background: #38853C;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
  background: #747474;
}

.emotion-0:focus:hover {
  background: #747474;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]):hover {
  background: #747474;
}

.emotion-0:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}

.emotion-0:checked {
  background: #38853C;
}

.emotion-0:checked:hover {
  background: #285F2B;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #ECF7EC,0rem 0rem 0rem 0.1875rem #83D187;
  background: #285F2B;
}

.emotion-0:checked:focus:hover {
  background: #285F2B;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #38853C;
}

.emotion-0:checked:focus:not([data-focus-visible-added]):hover {
  background: #285F2B;
}

.emotion-0:checked:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}



.emotion-0:checked:checked:after {
  -webkit-transform: translateX(1rem);
  -moz-transform: translateX(1rem);
  -ms-transform: translateX(1rem);
  transform: translateX(1rem);
}



.emotion-0:checked:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-0:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:after {
  background: #FFFFFF;
  content: "";
  display: block;
  position: absolute;
  border-radius: 1.25rem;
  -webkit-transition: -webkit-transform .25s ease;
  transition: transform .25s ease;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-3 >*:not(:first-child) {
  margin-top: 0rem;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #CC3D3D;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <input
        aria-checked="false"
        aria-label="label3"
        class="emotion-0"
        id="switch"
        role="switch"
        type="checkbox"
        value="switch3"
      />
      <div
        class="emotion-4"
      >
        <label
          class="emotion-1"
          for="switch"
        >
          Label
        </label>
        <div
          class="emotion-3"
        >
          <span
            class="emotion-2"
            role="alert"
          >
            Error text
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Switch tests all variants should match snapshot with help and error texts 1`] = `
<DocumentFragment>
  .emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-7 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-6 >* {
  margin-left: 0.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-5 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-0 {
  background: #B9B9B9;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  cursor: pointer;
  margin: unset;
  border-radius: 6.25rem;
  height: 1.25rem;
  width: 2.25rem;
}

.emotion-0:hover {
  background: #747474;
}

.emotion-0:active {
  background: #38853C;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
  background: #747474;
}

.emotion-0:focus:hover {
  background: #747474;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]):hover {
  background: #747474;
}

.emotion-0:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}

.emotion-0:checked {
  background: #38853C;
}

.emotion-0:checked:hover {
  background: #285F2B;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #ECF7EC,0rem 0rem 0rem 0.1875rem #83D187;
  background: #285F2B;
}

.emotion-0:checked:focus:hover {
  background: #285F2B;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #38853C;
}

.emotion-0:checked:focus:not([data-focus-visible-added]):hover {
  background: #285F2B;
}

.emotion-0:checked:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}



.emotion-0:checked:checked:after {
  -webkit-transform: translateX(1rem);
  -moz-transform: translateX(1rem);
  -ms-transform: translateX(1rem);
  transform: translateX(1rem);
}



.emotion-0:checked:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-0:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:after {
  background: #FFFFFF;
  content: "";
  display: block;
  position: absolute;
  border-radius: 1.25rem;
  -webkit-transition: -webkit-transform .25s ease;
  transition: transform .25s ease;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #CC3D3D;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <input
        aria-checked="true"
        aria-label="label2"
        checked=""
        class="emotion-0"
        id="switch"
        role="switch"
        type="checkbox"
      />
      <div
        class="emotion-5"
      >
        <label
          class="emotion-1"
          for="switch"
        >
          Label
        </label>
        <div
          class="emotion-4"
        >
          <span
            class="emotion-2"
          >
            Help text
          </span>
          <span
            class="emotion-3"
            role="alert"
          >
            Error text
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Switch tests all variants should match snapshot with help text 1`] = `
<DocumentFragment>
  .emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-6 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-5 >* {
  margin-left: 0.5rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-0 {
  background: #B9B9B9;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  cursor: pointer;
  margin: unset;
  border-radius: 6.25rem;
  height: 1.25rem;
  width: 2.25rem;
}

.emotion-0:hover {
  background: #747474;
}

.emotion-0:active {
  background: #38853C;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
  background: #747474;
}

.emotion-0:focus:hover {
  background: #747474;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]):hover {
  background: #747474;
}

.emotion-0:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}

.emotion-0:checked {
  background: #38853C;
}

.emotion-0:checked:hover {
  background: #285F2B;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #ECF7EC,0rem 0rem 0rem 0.1875rem #83D187;
  background: #285F2B;
}

.emotion-0:checked:focus:hover {
  background: #285F2B;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
  background: #38853C;
}

.emotion-0:checked:focus:not([data-focus-visible-added]):hover {
  background: #285F2B;
}

.emotion-0:checked:after {
  width: 0.75rem;
  height: 0.75rem;
  margin: 0.25rem;
}



.emotion-0:checked:checked:after {
  -webkit-transform: translateX(1rem);
  -moz-transform: translateX(1rem);
  -ms-transform: translateX(1rem);
  transform: translateX(1rem);
}



.emotion-0:checked:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-0:disabled:after {
  background: #747474;
  margin: 0.2rem;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:after {
  background: #FFFFFF;
  content: "";
  display: block;
  position: absolute;
  border-radius: 1.25rem;
  -webkit-transition: -webkit-transform .25s ease;
  transition: transform .25s ease;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-3 >*:not(:first-child) {
  margin-top: 0rem;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <input
        aria-checked="false"
        aria-label="label3"
        class="emotion-0"
        id="switch"
        role="switch"
        type="checkbox"
        value="switch3"
      />
      <div
        class="emotion-4"
      >
        <label
          class="emotion-1"
          for="switch"
        >
          Label
        </label>
        <div
          class="emotion-3"
        >
          <span
            class="emotion-2"
          >
            Help text
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
